articles/types/jsdoc.md

```diff
@@ -1,3 +1,179 @@
 # JsDoc support
 
-TODO
+Programs written in JavaScript and TypeScript can be annotated with (JsDoc comments)[http://usejsdoc.org/index.html]
+to give type information and quick-info descriptions of identifiers.
+
+In the following example, the text between the opening `/**` and closing `/*` are considered JsDoc comments.
+
+```js
+/**
+ * @param {"cat" | "dog"} kind  The type of animal to create.
+ * @param name {string}
+ * @param {number} weight The weight, in pounds, of animal.
+ * @param {...string} extraProps Any extra properties to tack on the animal.
+ * @return {{[index: string]: string | number}} Any extra properties to tack on the animal.
```

Comment:
This syntax is ... uh ... *extremely Typescript*. The JSDoc syntax is worse in my opinion (`Object.<string, string | number>`) but I think you should just rework the example to skip index signatures entirely.

Reply:
removed

Reply:
The object literal is better -- it's standard at least -- but still a little advanced for the first example. Probably fine though.